From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: JiaY-shi <shi05275@163.com>
Date: Mon, 31 Aug 2026 02:00:00 +0800
Subject: [PATCH] arm64: dts: mediatek: gl-mt2500: reserve firmware memory

The MT7981 BL31 firmware executes from 0x43000000. Keep Linux from
allocating that region, and retain the adjacent ramoops area used by
the vendor/OpenWrt memory map.

Signed-off-by: JiaY-shi <shi05275@163.com>

---
 .../dts/mediatek/mt7981b-glinet-gl-mt2500.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt7981b-glinet-gl-mt2500.dtsi b/arch/arm64/boot/dts/mediatek/mt7981b-glinet-gl-mt2500.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/mediatek/mt7981b-glinet-gl-mt2500.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt7981b-glinet-gl-mt2500.dtsi
@@ -31,6 +31,23 @@ memory@40000000 {
 		reg = <0 0x40000000 0 0x40000000>;
 	};
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		ramoops@42ff0000 {
+			compatible = "ramoops";
+			reg = <0 0x42ff0000 0 0x10000>;
+			record-size = <0x1000>;
+		};
+
+		secmon_reserved: secmon@43000000 {
+			reg = <0 0x43000000 0 0x30000>;
+			no-map;
+		};
+	};
+
 	gpio-keys {
 		compatible = "gpio-keys";
 
